src/Compilers/Core/Portable/CommandLine/AnalyzerConfigSet.cs

```diff
@@ -337,33 +337,40 @@ static void freeKey(List<Section> sectionKey, ObjectPool<List<Section>> pool)
 
         internal static bool TryParseSeverity(string value, out ReportDiagnostic severity)
         {
+            ReadOnlySpan<char> commentStartCharacters = stackalloc char[] { ';', '#' };
+            var trimmed = value.AsSpan();
+            var commentStartIndex = trimmed.IndexOfAny(commentStartCharacters);
+            if (commentStartIndex >= 0)
+                trimmed = trimmed[0..commentStartIndex].TrimEnd();
```

Comment:
As far as I understand, this change is to maintain the existing behavior and I assume we already have tests covering this. Is that right?

Reply:
➡️ Yes, and tests have now been added.